api/controllers/v2/cache.ts

```diff
@@ -0,0 +1,185 @@
+              name: 'REPOS',
+              value: JSON.stringify(repos),
+            },
+          ],
+        },
+      ],
+    },
+  });
+
+  await client.send(command);
+}
+
+/**
+ * Handles requests from individual doc sites and when the docs-worker-pool repository has a release with an updated Snooty Parser version.
+ * In the latter case, we should receive an event to build all doc site caches.
+ * @param {APIGatewayEvent} event An event object that comes from either a webhook payload or from the custom GitHub Action for the docs-worker pool.
+ *
+ * In either scenario, the body should contain an array of RepoInfo objects.
+ * @returns {Promise<APIGatewayProxyResult>}
+ */
+export async function rebuildCacheHandler(event: APIGatewayEvent): Promise<APIGatewayProxyResult> {
+  if (!event.body) {
+    const errorMessage = 'Error! No body found in event payload.';
+    console.error(errorMessage);
+    return {
+      statusCode: 400,
+      body: errorMessage,
+    };
+  }
+
+  const rebuildRequest = JSON.parse(event.body);
+
+  if (!isRebuildRequest(rebuildRequest)) {
+    const errorMessage = 'Error! Invalid rebuild request.';
+    console.error(errorMessage);
+    return {
+      statusCode: 400,
+      body: errorMessage,
+    };
+  }
+
+  try {
+    await runCacheRebuildJob(rebuildRequest);
+    return {
+      statusCode: 200,
+      body: 'Cache rebuild job successfully created',
+    };
+  } catch (e) {
+    console.error(e);
+    return {
+      statusCode: 500,
+      body: 'Error occurred when starting cache rebuild job',
+    };
+  }
+}
+
+/**
+ * This is for the GitHub webhooks. The GitHub webhooks will be used by individual doc sites to rebuild the cache if
+ * the snooty.toml file is modified.
+ * @param {APIGatewayEvent} event GitHub webhook push event. Body should be a PushEvent type.
+ * @returns {Promise<APIGatewayProxyResult>}
+ */
+export async function rebuildCacheGithubWebhookHandler(event: APIGatewayEvent): Promise<APIGatewayProxyResult> {
+  if (!event.body) {
+    const errorMessage = 'Error! No body found in event payload.';
+    console.error(errorMessage);
+    return {
+      statusCode: 400,
+      body: errorMessage,
+    };
+  }
+
+  let body: PushEvent;
+  try {
+    body = JSON.parse(event.body) as PushEvent;
+  } catch (e) {
+    console.log('[TriggerBuild]: ERROR! Could not parse event.body', e);
+    return {
+      statusCode: 502,
+      headers: { 'Content-Type': 'text/plain' },
+      body: ' ERROR! Could not parse event.body',
+    };
+  }
+
+  const repoOwner = body.repository.owner.login;
+  const repoName = body.repository.name;
+
+  // Checks the commits to see if there have been changes made to the snooty.toml file.
+  const snootyTomlChanged = body.commits.some(
+    (commit) =>
+      commit.added.some((fileName) => fileName === 'snooty.toml') ||
+      commit.removed.some((fileName) => fileName === 'snooty.toml') ||
+      commit.modified.some((fileName) => fileName === 'snooty.toml')
```

Comment:
Since we're looking for `snooty.toml` as the filename, does this mean that this won't be compatible with the monorepo?

Reply:
Yes, that is correct. The monorepo is not currently supported, but will be in the future